core/src/test/java/org/elasticsearch/plugins/PluginManagerTests.java

```diff
@@ -472,6 +480,53 @@ public void testOfficialPluginName_ThrowsException() throws IOException {
         }
     }
 
+    @Test
+    public void testThatBasicAuthIsSupported() throws Exception {
+        int port = randomIntBetween(49000, 65000);
```

Comment:
can we not do this and bind to port 0 (ephemeral) instead. Otherwise this test will always randomly fail, because i'm gonna sometimes have something running on that port.